drivers/sensor/nxp/fxls8974/fxls8974.h

```diff
@@ -26,7 +26,11 @@
 #define FXLS8974_REG_CTRLREG4				0x18
 #define FXLS8974_REG_CTRLREG5				0x19
 
+#ifdef CONFIG_FXLS8964
+#define WHOAMI_ID_FXLS8974				0x84
+#elif CONFIG_FXLS8964
 #define WHOAMI_ID_FXLS8974				0x86
+#endif /* CONFIG_FXLS8964 */
```

Comment:
if the only difference is this ID, maybe instead of a new compatible, a DT prop for ID makes sense? I'm fine with the compatible though, just think maybe this header file will have scaling issues as opposed to just using a DT prop for this value, ie if we need to make  a new #ifdef for every part number of the sensor family

Reply:
Sure, I have written a new approach using a dt prop instead.

Reply:
This would be quite opposite of most drivers here. I would want to get the OK from @MaureenHelm and @teburd on this approach.